File: examples/canonical-iac-pulumi/dotnet/azure/components/eventing-component/EventingComponent.cs
using Pulumi;
using Pulumi.AzureNative.ServiceBus;

public sealed class EventingComponent : ComponentResource
{
    public Output<string> QueueName { get; private set; }

    public EventingComponent(string name, string environment, string namePrefix, ComponentResourceOptions? opts = null)
        : base("accb:azure:EventingComponent", name, opts)
    {
        var stackName = Deployment.Instance.StackName;
        var queue = new Queue($"{namePrefix}-{environment}-{stackName}-main", new QueueArgs
        {
            MaxDeliveryCount = 5
        }, new CustomResourceOptions { Parent = this });
        QueueName = queue.Name;
        RegisterOutputs(new Dictionary<string, object?> { ["queueName"] = QueueName });
    }
}

File: examples/canonical-iac-pulumi/python/azure/components/eventing_component.py
import pulumi


class EventingComponent(pulumi.ComponentResource):
    def __init__(self, name: str, environment: str, name_prefix: str, opts=None):
        super().__init__("accb:azure:EventingComponent", name, None, opts)
        stack = pulumi.get_stack()
        self.eventing_name = pulumi.Output.from_input(f"{name_prefix}-{environment}-{stack}-events")
        self.register_outputs({"eventing_name": self.eventing_name})
